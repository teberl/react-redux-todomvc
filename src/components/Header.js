import React from 'react';
import PropTypes from 'prop-types';
import TodoTextInput from './TodoTextInput';

Header.propTypes = {
  addTodo: PropTypes.func.isRequired
};

function Header({ addTodo }) {
  return (
    <header className="header">
      <h1>todos</h1>
      <TodoTextInput
        newTodo
        placeholder="What needs to be done?"
        onSave={text => addTodo(text)}
      />
    </header>
  );
}

export default Header;
