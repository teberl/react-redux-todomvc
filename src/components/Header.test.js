import React from 'react';
import TestRenderer from 'react-test-renderer';
import Header from './Header';

it('renders correctly', () => {
  const testRenderer = TestRenderer.create(<Header />).toJSON();

  expect(testRenderer).toMatchSnapshot();
});
