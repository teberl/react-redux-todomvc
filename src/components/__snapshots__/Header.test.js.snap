// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correct 1`] = `
<header
  className="header"
>
  <h1>
    todos
  </h1>
  <input
    autoFocus={true}
    className="edit new-todo"
    onBlur={[Function]}
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="What needs to be done?"
    type="text"
    value=""
  />
</header>
`;
